cluster-provision/gocli/pkg/libssh/ssh.go

```diff
@@ -276,3 +240,24 @@ func (s *SSHClientImpl) CopyRemoteFile(remotePathToCopy string, target io.Writer
 
 	return err
 }
+
+func (s *SSHClientImpl) initClient() error {
```

Comment:
This is not thread safe, you should ensure that the initialization is synchronized.

A naive implementation would look like this:

```suggestion
var initMutex sync.Mutex

func (s *SSHClientImpl) initClient() error {
	initMutex.Lock()
	defer initMutex.Unlock()
	if s.client != nil {
		return nil
	}
```

Reply:
Apologies, my bad - the initMutex should be part of the instance struct

Reply:
@dhiller okay let me implement this and i will ping you for a retest shortly